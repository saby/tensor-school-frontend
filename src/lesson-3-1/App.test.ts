import { configureToMatchImageSnapshot } from 'jest-image-snapshot';
import puppeteer, { Page, Browser } from 'puppeteer';
import path from 'path';

const customConfig = { threshold: 0 };
const toMatchImageSnapshot = configureToMatchImageSnapshot({
    customDiffConfig: customConfig,
    noColors: true
});
expect.extend({ toMatchImageSnapshot });

let browser: Browser;

beforeAll(async () => {
    browser = await puppeteer.launch({
        headless: true
    });
});

afterAll(async () => {
    await browser.close();
});

describe('Сетка', () => {
    let page: Page;
    beforeEach(async () => {
        page = await browser.newPage();
        await page.goto(`file:${path.join(__dirname, 'index.html')}`);
    });

    it('renders correctly', async () => {
        const image = await page.screenshot();

        expect(image).toMatchImageSnapshot();
    });

    it('Верстка соответствует эталону', async () => {
        const container = await page.evaluate(() => document.body.innerHTML);

        expect(container).toMatchSnapshot();
    });
});
