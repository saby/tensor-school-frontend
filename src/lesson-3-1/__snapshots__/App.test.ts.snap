// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Сетка Верстка соответствует эталону 1`] = `
"
    <div class=\\"container\\">
        <header class=\\"header\\">
            <a href=\\"#\\" class=\\"header__link\\">
                <img class=\\"header__image\\" src=\\"img/tensor-logo.png\\" alt=\\"Главная страница\\">
            </a>
            <h1 class=\\"header__title\\">
                Tensor school
            </h1>
        </header>
        <main>
            <figure class=\\"school-logo\\">
                <img src=\\"img/school-logo.svg\\" alt=\\"Логотип школы\\" class=\\"school-logo__image\\">
                <figcaption class=\\"school-logo__caption\\">
                    <p class=\\"school-logo__paragraph\\">
                        Это страница школы Тензор.
                    </p>
                    <p class=\\"school-logo__paragraph\\">
                        Тут вы можете познакомиться с нашими учениками и посмотреть темы занятий.
                    </p>
                </figcaption>
            </figure>
        </main>
    </div>

"
`;
